pkg/apis/kai/v1/pod_grouper/pod_grouper.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+// +kubebuilder:object:generate:=true
+package pod_grouper
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/ptr"
+
+	"github.com/NVIDIA/KAI-scheduler/pkg/apis/kai/v1/common"
+)
+
+const (
+	imageName = "pod-grouper"
+)
+
+type PodGrouper struct {
+	// Enabled defines whether the pod grouper should be deployed
+	// +kubebuilder:validation:Optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Image is the configuration of the pod-grouper image
```

Comment:
pod-grouper again

Just kidding 

But make it uniform with the dash - either do `pod-grouper` in all, or do `pod grouper`